### PR TITLE
Collections project item creation

### LINKED ISSUE #7727: There shouldn't be two menus for creating new artifacts
This is in the collections project branch. The plan is to turn the "New" button in the top nav back to just "New question," and to have the plus icon on collections display a menu with new dashboard, new pulse, and new collection. The latter is present in the UI, but the actions are not wired up.

screen shot 2018-05-25 at 10 58 40 am

### PR DESCRIPTION
**Pull request by kdoh**
Moves item creation (dashboard, pulses, etc) to collection pages rather than the top nav. Resolves #7727  and #7750

### CODE REVIEW

**Comment by mazameli** [comment]
Everything looks good except a few funny things with canceling:

1. If you try and x out of the new dashboard modal, you get a 404. Also seems like the New Dashboard modal should have a cancel button instead of an x so that it's the same as the New Collection modal.
2. If you cancel a new pulse, you go to the old pulse list page (/pulse). I can open a separate issue for this.
3. You can't cancel or click outside of the New Collection modal. I don't see any JS errors, weirdly.

Unrelated, it looks like these two buttons have different type sizes:

screen shot 2018-05-29 at 3 36 58 pm

**Comment by kdoh** [comment]
@mazameli 

1. The 404 should be fixed now, also added the cancel button for dashboard creation.
2. Pulses now redirect to the collection that triggered the new pulse creation

@tlrobinson It seems like both of these have a `OnClickOutsideWrapper` but clicks or `esc` key hits aren't triggering closes. Is there anything that needs to be specified to ensure this happens?

File: frontend/src/metabase/components/CreateDashboardModal.jsx [hunk 1]
```diff
@@ -127,7 +142,10 @@ export default class CreateDashboardModal extends Component {
                     errors={this.state.errors}
                   >
                     <Select
-                      value={this.state.collection_id}
+                      value={
+                        this.state.collection_id ||
+                        this.props.params.collectionId
```
Comment (kdoh) [inline frontend/src/metabase/components/CreateDashboardModal.jsx]: @tlrobinson Is there a good example somewhere of how to populate an initial value on selects? I'm running into issues with both the dashboard modal and the pulse collection selection when I pass the context of the current collection in.

Comment (tlrobinson) [reply frontend/src/metabase/components/CreateDashboardModal.jsx]: Generally we'd set the default state from the props in the constructor, i.e.

```javascript
constructor(props) {
  super(props);
  this.state = {
    collectionId: props.params.collectionId
  }
}
```

Of course that requires `collectionId` be a param on the current route (not just the "referring" route). Is that the case?

**Comment by kdoh** [comment]
Alrighty I think all the major things here should have been addressed.

**Pull request merged**